### DISCUSSION TITLE
"How can I create a PRC diagram plot using macroinvertebrate community data?

### DISCUSSION THREAD

**Discussion by AnnaRiver94** [community/community#123765] [Other Feature Feedback, Questions, & Ideas]
### R code

I have this dataframe:
$ Stretch         : Factor w/ 3 levels "UP","TAIL","DOWN": 1 2 3 1 2 3 1 2 3 1 ...
 $ Cod             : chr [1:36] "S1" "S2" "S3" "S1" ...
 $ Month           : num [1:36] 8 8 8 10 10 10 6 6 6 2 ...
 $ Season          : chr [1:36] "s" "s" "s" "a" ...
 $ Year            : num [1:36] 2018 2018 2018 2018 2018 ...
 $ Date            : POSIXct[1:36], format: "2018-08-01" "2018-08-01" "2018-08-01" ...
 $ Impact          : chr [1:36] "Before" "Before" "Before" "Before" ...
 $ Sampling_dates  : Factor w/ 12 levels "T1","T10","T11",..: 1 1 1 5 5 5 6 6 6 7 ...
 $ Dates           : num [1:36] 1 1 1 2 2 2 3 3 3 4 ...
 $ meso habitat    : chr [1:36] "G" "G" "G" "G" ...
 $ Q               : num [1:36] 2.55 2.55 2.55 1.3 1.3 ...
 $ N               : num [1:36] 644 645 1273 398 615 ...
 $ S               : num [1:36] 14 18 18 17 17 19 20 21 19 8 ...
 $ BAETIDAE        : num [1:36] 7 44 49 28 10 16 11 50 159 1 ...
 $ CAENIDAE        : num [1:36] 4 1 141 1 51 14 7 2 2 0 ...
 $ CERATOPOGONIDAE : num [1:36] 3 1 10 1 2 0 10 7 0 2 ...
 $ CHIRONOMIDAE    : num [1:36] 25 19 328 43 54 21 32 87 47 421 ...
 $ DRYOPIDAE       : num [1:36] 0 6 22 6 4 6 1 0 10 0 ...
 $ DUGESIIDAE      : num [1:36] 5 32 0 26 20 8 0 9 12 0 ...
 $ ELMIDAE         : num [1:36] 35 61 130 20 64 7 33 101 22 1 ...
 $ EMPIDIDAE       : num [1:36] 0 1 2 1 2 0 1 1 1 0 ...
 $ EPHEMERELLIDAE  : num [1:36] 0 0 19 0 0 0 72 118 130 0 ...
 $ GOMPHIDAE       : num [1:36] 3 14 16 24 38 27 0 5 13 0 ...
 $ HELODIDAE       : num [1:36] 0 0 0 25 0 2 12 6 0 0 ...
 $ HEPTAGENIIDAE   : num [1:36] 0 0 1 3 0 3 1 5 4 0 ...
 $ HYDRACARINA     : num [1:36] 334 151 133 93 118 138 242 526 373 2 ...
 $ HYDROPSYCHIDAE  : num [1:36] 77 200 214 65 159 59 6 82 38 0 ...
 $ LEPTOCERIDAE    : num [1:36] 23 24 82 12 9 2 19 64 54 0 ...
 $ LEPTOPHLEBIIDAE : num [1:36] 58 3 20 0 1 2 10 6 4 1 ...
 $ LEUCTRIDAE      : num [1:36] 63 18 61 21 62 35 290 249 85 2 ...
 $ LIMONIIDAE      : num [1:36] 3 3 22 0 0 2 1 7 0 0 ...
 $ NAIDIDAE        : num [1:36] 0 0 16 27 10 8 11 2 7 0 ...
 $ NEMATODA        : num [1:36] 0 1 0 0 0 17 1 0 2 1 ...
 $ RHYACOPHILIDAE  : num [1:36] 0 0 0 0 0 0 0 21 8 0 ...
 $ SIMULIIDAE      : num [1:36] 0 60 0 0 1 1 2 10 13 0 ...
 $ TAENIOPTERYGIDAE: num [1:36] 0 0 0 0 0 0 0 0 0 0 ...
 $ TIPULIDAE       : num [1:36] 4 6 7 2 10 3 3 6 0 0 ...
I applied this function for PRC
 mod <- prc(response = cpio, treatment = piota$Stretch, time = piota$Sampling_dates)
cpio is a subset of community data (from BAETIDAE to TIPULIDAE) . But when I plotted my graph I had this error 
Messaggi di avvertimento:
1: In xy.coords(x, y, xlabel, ylabel, log = log, recycle = TRUE) :
  NA introdotti per coercizione
2: In xy.coords(x, y, xlabel, ylabel, log) : NA introdotti per coercizione
3: In xy.coords(x, y) : NA introdotti per coercizione
4: In rug(x$sp, side = 4) : alcuni valori saranno tagliati
I would like to create a plot as in the image
![immagine_2024-05-14_132334771](https://github.com/community/community/assets/169799074/dd5c65d4-37b6-4e42-9eba-41de3b9864da)

**Comment by nikhilsbuilds** [comment]
To create a plot like the one in the image, you can use the `plot` function provided by the `prc` package. Here's an example code:

```r
# Load the required package
library(vegan)

# Plot the PRC result
plot(mod, select = 1:3, scaling = "max", ylim = c(-1.5, 1.5), xlab = "Sampling Dates", 
     ylab = "Community Weight", cex.lab = 1.2, cex.axis = 1)

# Add a legend
legend("topright", legend = levels(piota$Stretch), lty = 1, col = 1:3, 
       cex = 0.8, bty = "n")
```

Here's what each argument in the `plot` function does:

- `select = 1:3`: This selects the treatment levels to be plotted. In your case, there are three levels of `Stretch` ("UP", "TAIL", and "DOWN"), so you can specify `1:3` to plot all three levels.
- `scaling = "max"`: This scales the y-axis to the maximum value, such that the highest community weight is set to 1.
- `ylim = c(-1.5, 1.5)`: This sets the limits of the y-axis from -1.5 to 1.5.
- `xlab` and `ylab`: These set the labels for the x and y axes, respectively.
- `cex.lab` and `cex.axis`: These control the font size for the axis labels and axis tick labels, respectively.

The `legend` function is used to add a legend to the plot, where `legend` specifies the legend labels, `lty` sets the line type, `col` sets the colors for the lines, `cex` controls the font size of the legend text, and `bty = "n"` removes the border around the legend.

Make sure to replace `piota$Stretch` and `piota$Sampling_dates` with the names of your corresponding columns in the data frame.

Note: If you encounter any warnings or errors related to missing values (`NA`), you may need to handle them appropriately, such as removing rows with missing values or imputing them with appropriate values.
